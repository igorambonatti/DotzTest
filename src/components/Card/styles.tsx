import styled from 'styled-components/native'

export const Container = styled.View`
  border: 1px solid #fff;
  border-radius: 10px;
  justify-content: center;
  margin: 12px;
  padding: 20px;
  height: 100px;
`
export const IconView = styled.View`
  margin-top: 4px;
`
export const Title = styled.Text`
  font-size: 18px;
  color: #f4ede8;
  margin-left: auto;
  margin-top: 15px;
  text-align: right;
  height: 80%;
`

export const ButtonSelect = styled.TouchableOpacity`
  width: 50%;
`
