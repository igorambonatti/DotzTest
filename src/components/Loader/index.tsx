import React from 'react'
import { View, ActivityIndicator } from 'react-native'

// import { Container } from './styles';

const Loader: React.FC = () => {
  return (
    <View
      style={{
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center',
        backgroundColor: '#F56700',
      }}
    >
      <ActivityIndicator size='large' color='#fff' />
    </View>
  )
}

export default Loader
