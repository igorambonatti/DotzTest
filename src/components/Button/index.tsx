import React from 'react'
import { RectButtonProperties } from 'react-native-gesture-handler'

import { Container, ButtonText } from './styles'

export interface ButtonProps extends RectButtonProperties {
  children: string
}

const Button: React.FC<ButtonProps> = ({ children, ...rest }) => (
  <Container testID={'button'} {...rest}>
    <ButtonText>{children}</ButtonText>
  </Container>
)

export default Button
