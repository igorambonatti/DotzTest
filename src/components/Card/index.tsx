import React from 'react'
import { View, Text } from 'react-native'
import { ViewProps } from 'react-native'
import FontAwesome from 'react-native-vector-icons/FontAwesome'

import * as S from './styles'

export interface CardProps extends ViewProps {
  icon: string
  title: string
  onSelect: () => void
}

const Card: React.FC<CardProps> = ({ icon, title, onSelect, ...rest }) => {
  return (
    <S.ButtonSelect onPress={onSelect} testID={'button'}>
      <S.Container {...rest}>
        <S.IconView>
          <FontAwesome name={icon} size={24} color='#fff' />
        </S.IconView>
        <S.Title testID={'title'}>{title}</S.Title>
      </S.Container>
    </S.ButtonSelect>
  )
}

export default Card
