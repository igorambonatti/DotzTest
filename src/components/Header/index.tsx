import React from 'react'
import FontAwesome from 'react-native-vector-icons/FontAwesome'

import { Body, Header as HeaderNB, Left, Right } from 'native-base'

import * as S from './styles'

export interface HeaderProps {
  title?: string
  onBack?: () => void
}

const Header: React.FC<HeaderProps> = ({ onBack, title, ...rest }) => {
  return (
    <S.Container
      androidStatusBarColor='#FFF'
      iosBarStyle='dark-content'
      noShadow
      {...rest}
      testID={'container'}
    >
      <Left>
        <S.BackButton onPress={onBack}>
          <FontAwesome name='angle-left' size={36} color='#fff' />
        </S.BackButton>
      </Left>
      <Body
        style={{
          flex: Platform.OS !== 'ios' ? 6 : 1,
          alignItems: 'center',
          left: Platform.OS !== 'ios' ? 20 : 0,
        }}
      >
        <S.Title>{title}</S.Title>
      </Body>
      <Right />
    </S.Container>
  )
}

export default Header
