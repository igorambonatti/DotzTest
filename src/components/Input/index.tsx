import React, {
  useEffect,
  useCallback,
  useRef,
  useImperativeHandle,
  forwardRef,
  useState,
} from 'react'
import { TextInputProps, Text } from 'react-native'

import { useField } from '@unform/core'

import * as S from './styles'

export interface InputProps extends TextInputProps {
  name: string
}

interface InputValueReference {
  value?: string
}

interface InputRef {
  focus(): void
}

const Input: React.FC<InputRef, InputProps> = (
  { name, icon, containerStyle = {}, ...rest },
  ref,
) => {
  const inputElementRef = useRef<any>(null)

  const { registerField, defaultValue = '', fieldName, error } = useField(name)
  const inputValueRef = useRef<InputValueReference>({ value: defaultValue })

  const [isFocused, setIsFocused] = useState(false)
  const [isFilled, setIsFilled] = useState(false)

  const handleInputFocus = useCallback(() => {
    setIsFocused(true)
  }, [])

  const handleInputBlur = useCallback(() => {
    setIsFocused(false)
    setIsFilled(!!inputValueRef.current.value)
  }, [])

  useImperativeHandle(ref, () => ({
    focus() {
      inputElementRef.current.focus()
    },
  }))

  useEffect(() => {
    registerField<string>({
      name: fieldName,
      ref: inputValueRef.current,
      path: 'value',
    })
  }, [fieldName, registerField])

  return (
    <>
      {error && <S.LabelError>{error}</S.LabelError>}
      <S.Container
        style={containerStyle}
        isFocused={isFocused}
        isErrored={!!error}
      >
        <S.TextInput
          testID={'input'}
          ref={inputElementRef}
          keyboardAppearance='dark'
          placeholderTextColor='#fff'
          defaultValue={defaultValue}
          onFocus={handleInputFocus}
          onBlur={handleInputBlur}
          onChangeText={(value) => {
            inputValueRef.current.value = value
          }}
          {...rest}
        />
      </S.Container>
    </>
  )
}

export default forwardRef(Input)
