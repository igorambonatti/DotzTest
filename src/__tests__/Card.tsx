import React from 'react'
import { create, act } from 'react-test-renderer'

import { CardProps } from '../components/Card'
import Card from '../components/Card'

let props: CardProps = {
  icon: 'usd',
  title: 'Dotz',
  onSelect: jest.fn(),
}
jest.mock('react-native-vector-icons/FontAwesome', () => 'usd')
const tree = create(<Card {...props} />)

test('snapshot', () => {
  expect(tree).toMatchSnapshot()
})

it(`render component`, async () => {
  const card = tree.root.findByProps({ testID: 'button' }).props
  expect(card).toBeTruthy()
})
it(`render component `, async () => {
  const card = tree.root.findByProps({ testID: 'title' }).props
  expect(card).toBeTruthy()
})
test('button press', () => {
  const card = tree.root.findByProps({ testID: 'button' }).props
  act(() => card.onPress())
})
