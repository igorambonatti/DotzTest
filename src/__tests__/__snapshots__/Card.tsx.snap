// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
      "width": "50%",
    }
  }
  testID="button"
>
  <View
    style={
      Array [
        Object {
          "borderColor": "#fff",
          "borderRadius": 10,
          "borderStyle": "solid",
          "borderWidth": 1,
          "height": 100,
          "justifyContent": "center",
          "marginBottom": 12,
          "marginLeft": 12,
          "marginRight": 12,
          "marginTop": 12,
          "paddingBottom": 20,
          "paddingLeft": 20,
          "paddingRight": 20,
          "paddingTop": 20,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "marginTop": 4,
          },
        ]
      }
    >
      <usd
        color="#fff"
        name="usd"
        size={24}
      />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#f4ede8",
            "fontSize": 18,
            "height": "80%",
            "marginLeft": "auto",
            "marginTop": 15,
            "textAlign": "right",
          },
        ]
      }
      testID="title"
    >
      Dotz
    </Text>
  </View>
</View>
`;
